### PR TITLE
feat: support pgvecto.rs as a VectorStore

### PR DESCRIPTION
**Pull request by BeautyyuYanli**
# Description

Support [pgvecto.rs]() as a VectorStore

Relate to https://github.com/tensorchord/pgvecto.rs/issues/98

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?

As described in the new notebook.

- [ ] Added new unit/integration tests
- [x] Added new notebook (that tests end-to-end)
- [ ] I stared at the code and made sure it makes sense

# Suggested Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added Google Colab support for the newly added notebooks.
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

File: docs/examples/vector_stores/PGVectoRsDemo.ipynb [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+ "cells": [
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "307804a3-c02b-4a57-ac0d-172c30ddc851",
+   "metadata": {},
+   "source": [
+    "# pgvecto.rs\n",
+    "\n",
+    "<a href=\"https://colab.research.google.com/github/run-llama/llama_index/blob/main/docs/examples/vector_stores/PineconeIndexDemo.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>"
```
Comment (hatianzhang) [inline docs/examples/vector_stores/PGVectoRsDemo.ipynb:11]: nitpick: wrong file name for colab

File: llama_index/vector_stores/pgvecto_rs.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    def add(
+        self,
+        nodes: List[BaseNode],
+    ) -> List[str]:
+        from pgvecto_rs.sdk import Record
+
+        records = [
+            Record(
+                id=node.id_,
+                text="",
```
Comment (hatianzhang) [inline llama_index/vector_stores/pgvecto_rs.py:49]: is the text here empty?

Comment (BeautyyuYanli) [reply llama_index/vector_stores/pgvecto_rs.py:49]: Yes, the text field is not needed for  llamaindex

File: llama_index/vector_stores/pgvecto_rs.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+                embedding=node.get_embedding(),
+            )
+            for node in nodes
+        ]
+
+        self._client.insert(records)
+        return [node.id_ for node in nodes]
+
+    def delete(self, ref_doc_id: str, **delete_kwargs: Any) -> None:
+        self._client.delete_by_ids((ref_doc_id,))
```
Comment (logan-markewich) [inline llama_index/vector_stores/pgvecto_rs.py:60]: Quick note -- this isn't actually deleting by ids, it should be deleting every node that has a matching `ref_doc_id` in it's metadata

Is this possible?

Comment (logan-markewich) [reply llama_index/vector_stores/pgvecto_rs.py]: Figured it out :)

Comment (BeautyyuYanli) [reply llama_index/vector_stores/pgvecto_rs.py]: Thank you for your help!

**Pull request merged**